core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java

```diff
@@ -303,10 +306,8 @@ private List<String> appendsAfterScan(long fromSnapshotId) {
     return filesToScan(appendsAfter);
   }
 
-  private List<String> appendsBetweenScan(long fromSnapshotId, long toSnapshotId) {
-    Snapshot s1 = table.snapshot(fromSnapshotId);
-    Snapshot s2 = table.snapshot(toSnapshotId);
-    TableScan appendsBetween = table.newScan().appendsBetween(s1.snapshotId(), s2.snapshotId());
+  private List<String> appendsBetweenScan(Long fromSnapshotId, long toSnapshotId) {
+    TableScan appendsBetween = table.newScan().appendsInRange(fromSnapshotId, toSnapshotId);
     return filesToScan(appendsBetween);
   }
```

Comment:
How about adding another method, and keeping the original?
```suggestion

  private List<String> appendsBeforeScan(long toSnapshotId) {
    TableScan appendsBetween = table.newScan().appendsInRange(null, toSnapshotId);
    return filesToScan(appendsBetween);
  }

  private List<String> appendsBetweenScan(long fromSnapshotId, long toSnapshotId) {
    TableScan appendsBetween = table.newScan().appendsInRange(fromSnapshotId, toSnapshotId);
    return filesToScan(appendsBetween);
  }
```

Reply:
This change seems unnecessary. let me try to revert it and minimize the change in this test class

Reply:
Are there still cases where startSnapshotId is `null` after reverting?

Reply:
Not in the existing unit test of `TestIncrementalDataTableScan`. But this reminds me that maybe we should add unit test with null startSnapshotId for the `appendsInRange`. But we do not need to change this method. if there is a need, we can create a new method as you suggested.

Reply:
it's better now from my side. thanks